src/contracts/proxies/actions/BasicActions.sol

```diff
@@ -178,20 +159,14 @@ contract BasicActions is CommonActions, IBasicActions {
   ) internal {
     address _safeEngine = ODSafeManager(_manager).safeEngine();
     ODSafeManager.SAFEData memory _safeInfo = ODSafeManager(_manager).safeData(_safeId);
-    ITaxCollector(_taxCollector).taxSingle(_safeInfo.collateralType);
 
     // Takes token amount from user's wallet and joins into the safeEngine
     _joinCollateral(_collateralJoin, _safeInfo.safeHandler, _collateralAmount);
 
     int256 deltaDebt = _getGeneratedDeltaDebt(_safeEngine, _safeInfo.collateralType, _safeInfo.safeHandler, _deltaWad);
 
     // Locks token amount into the SAFE and generates debt
-    _modifySAFECollateralization(
-      _manager,
-      _safeId,
-      _collateralAmount.toInt(),
-      deltaDebt
-    );
+    _modifySAFECollateralization(_manager, _safeId, _collateralAmount.toInt(), deltaDebt, false);
```

Comment:
Is there any impact to moving `taxSingle` via `_modifySAFECollateralization ` to _after_ `_getGeneratedDeltaDebt`, `_joinCollateral`?

Reply:
good question! I did a bit of digging.

This change moves taxCollector.taxSingle from before `_joinCollateral ` and `_getGeneratedDeltaDebt`.

Therefore, we must check whether taxSingle impacts any state which `_joinCollateral ` and `_getGeneratedDeltaDebt` might read from or vice versa.

That is, we want to check if the order of operations are important due to dependency of state from either side.

**taxCollector.taxSingle**
- might call `safeEngine.updateAccumulatedRate` via `_splitTaxIncome` via `_distributeTax` 
  - This updates `__cData.accumulatedRate` in `SafeEngine` 
  - `_modifyInternalCoins` is also called in `safeEngine.updateAccumulatedRate` and `coinBalance[_dst]` and `globalDebt` are modified

**BasicActions._joinCollateral**
- transfers some token from the msg.sender to the contract
- approve collateral join contract and call collateralJoin.join which calls `safeEngine.modifyCollateralBalance`
  - this modifies: `tokenCollateral[_cType][_account]`

**BasicActions._getGeneratedDeltaDebt**
- This function retrieves: `ISAFEEngine(_safeEngine).coinBalance(_safeHandler);` and `ISAFEEngine(_safeEngine).cData(_cType).accumulatedRate`
- These are both modified in the `taxSingle` function, which previously did the following:
  - accumulated rate is increased (assuming `_currentTaxCut` is always positive)
  - `_modifyInternalCoins` also increments coinBalance

This means that moving tax single after will result in a larger value from `_getGeneratedDeltaDebt` after this change. Part of `_getGeneratedDeltaDebt`:
```
uint256 _rate = ISAFEEngine(_safeEngine).cData(_cType).accumulatedRate;
uint256 _coinAmount = ISAFEEngine(_safeEngine).coinBalance(_safeHandler);
 // Calculates the needed deltaDebt so together with the existing coins in the safeEngine is enough to exit wad amount of COIN tokens
 _deltaDebt = ((_deltaWad * RAY - _coinAmount) / _rate).toInt();
 // This is neeeded due lack of precision. It might need to sum an extra deltaDebt wei (for the given COIN wad amount)
 _deltaDebt = uint256(_deltaDebt) * _rate < _deltaWad * RAY ? _deltaDebt + 1 : _deltaDebt;
```

Previously `taxSingle` would increase the value of `_rate` and `_coinAmount` which would reduce the `_deltaDebt` value.

I can't say for certain what impact this would have on the system-it would be a lot easier to see if there are some issues with the changes if the tests were working.